### PR TITLE
removed llen from redis command pipe-line as it seems to be reduandent

### PR DESCRIPTION
**Pull request by auvipy**
To fix https://github.com/celery/celery/issues/4874 

@georgepsarakis should the commands from tests need to be removed too?
@bohea

### CODE REVIEW

**Comment by auvipy** [comment]
unit test is failing for the code change

**Comment by georgepsarakis** [comment]
@auvipy this is not a valid change. The `llen` command is used to count how many tasks have completed, thus detecting if all the chord header tasks have finished, and then running the chord callback. It is therefore required.

File: celery/backends/redis.py [hunk 1]
```diff
@@ -346,7 +346,6 @@ def on_chord_part_return(self, request, state, result,
         with client.pipeline() as pipe:
             _, readycount, totaldiff, _, _ = pipe \
```
[the remark this answers is not in the document]
Comment (auvipy) [reply celery/backends/redis.py]: what adjustment is needed in tests?

[the remark this answers is not in the document]
Comment (georgepsarakis) [reply celery/backends/redis.py]: The mock Redis client seems to have an inconsistent return value for [rpush](https://github.com/celery/celery/blob/master/t/unit/backends/test_redis.py#L99).

I think this method needs to be changed to:

```python
key_list = self._get_list(key)
key_list.append(value)
return len(key_list)
```

**Comment by thedrow** [comment]
@auvipy Will you continue to work on this?

**Comment by auvipy** [comment]
you can take over if you have interest.

**Comment by georgepsarakis** [comment]
@auvipy I did some minor fixes, I hope it is not a problem. As a side-note, we should squash the commits before merging.

**Comment by auvipy** [comment]
george, any kind of collaboration is always welcomed :D no worries mate!

**Pull request closed without merging**